client/core/core.go

```diff
@@ -4979,6 +5018,72 @@ func (c *Core) PreOrder(form *TradeForm) (*OrderEstimate, error) {
 
 // Trade is used to place a market or limit order.
 func (c *Core) Trade(pw []byte, form *TradeForm) (*Order, error) {
+	dc, crypter, err := c.parseTradeCredentials(pw, form)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := c.prepareTrackedTrade(dc, form, crypter)
```

Comment:
There no longer seems to be a reason to factor `parseTradeCredentials` out of `prepareTrackedTrade`.

However, an unintended side effect is that `crypter.Close()` seems to run far too soon now that it is deferred in `parseTradeCredentials` helper instead of after the crypter is finished in `prepareTrackedTrade`.

Maybe eliminate `parseTradeCredentials` and do this:

```diff
@@ -5026,12 +5030,7 @@ func (c *Core) PreOrder(form *TradeForm) (*OrderEstimate, error) {
 
 // Trade is used to place a market or limit order.
 func (c *Core) Trade(pw []byte, form *TradeForm) (*Order, error) {
-       dc, crypter, err := c.parseTradeCredentials(pw, form)
-       if err != nil {
-               return nil, err
-       }
-
-       req, err := c.prepareTrackedTrade(dc, form, crypter)
+       req, err := c.prepareTrackedTrade(form, pw)
        if err != nil {
                return nil, err
        }
@@ -5048,15 +5047,11 @@ func (c *Core) Trade(pw []byte, form *TradeForm) (*Order, error) {
 // server validation. This helps handle some issues related to UI/UX where
 // server response might take a fairly long time (15 - 20s).
 func (c *Core) TradeAsync(pw []byte, form *TradeForm) (*InFlightOrder, error) {
-       dc, crypter, err := c.parseTradeCredentials(pw, form)
-       if err != nil {
-               return nil, err
-       }
-
-       req, err := c.prepareTrackedTrade(dc, form, crypter)
+       req, err := c.prepareTrackedTrade(form, pw)
        if err != nil {
                return nil, err
        }
+       dc := req.dc
 
        // Prepare and store the inflight order.
        corder := coreOrderFromTrade(req.dbOrder.Order, req.dbOrder.MetaData)
@@ -5131,7 +5102,27 @@ type tradeRequest struct {
 }
 
 // prepareTrackedTrade prepares a trade request.
-func (c *Core) prepareTrackedTrade(dc *dexConnection, form *TradeForm, crypter encrypt.Crypter) (*tradeRequest, error) {
+func (c *Core) prepareTrackedTrade(form *TradeForm, pw []byte) (*tradeRequest, error) {
... <paste contents of parseTradeCredentials> 
```